lib/puppet/util/tagging.rb

```diff
@@ -1,26 +1,20 @@
 require 'puppet/util/tag_set'
 
 module Puppet::Util::Tagging
-  ValidTagRegex = /^\w[-\w:.]*$/
-
-  # Add a tag to the current tag set.
-  # When a tag set is used for a scope, these tags will be added to all of
-  # the objects contained in this scope when the objects are finished.
-  #
+  # Add a tag to our current list.  These tags will be added to all
+  # of the objects contained in this scope.
   def tag(*ary)
     @tags ||= new_tags
 
-    ary.flatten.each do |tag|
-      name = tag.to_s.downcase
-      if name =~ ValidTagRegex
-        @tags << name
-        name.split("::").each do |section|
-          @tags << section
-        end
```

Comment:
These three lines should be conditional on the return of a new `split_qualified_tags?` method. This method returns `true` for this class. In the Transaction class, where it used to override `handle_qualified_tags`, override the `split_qualified_tags?` method and return `false`